### ISSUE TITLE
Hostname Verification Failure

### ISSUE THREAD

**Issue by PapistPenguin** [itprojects/InboxPager#30]
I tried setting up my email, but when I clicked "Refresh All Inboxes," it showed "Action failed (see log file).""Errors / Exceptions" showed "nullException:'mail.encavo.net' != 'null'" and "Exception:'mail.encavo.net' != 'null'". I have been able to setup my email on other email clients. I do not know why this one is problematic.

**Comment by itprojects** [comment]
It could be a broken configuration.

Do you know your server settings?
Example: imap.encavo.net:993 or smtp.encavo:465
Did you use this format: user@encavo.net or just user without @encavo.net?
Does this server allow SSL-apps to connect, or do you have to enable through your browser. This would typically be the case for gmail.

**Comment by itprojects** [comment]
> I have been able to setup my email on other email clients.

@PapistPenguin Can you share an anonymized configuration for this inbox, as used in another app? (If yes, please redact username and password!)

**Comment by itprojects** [comment]
@PapistPenguin Without more information from you, this bug will be closed, because Inbox Pager does work with e-mail servers.

**Comment by oizo** [comment]
It seems I'm experiencing the same issue. 
I get a dialog with the text:
`'mail.hvam.io' != 'null'
net.inbox.server.socketIO:113
Java.lang.Thread:919`

The server settings are working in other clients, why I would assume it's a bug in inbox

**Comment by oizo** [comment]
It seems the SSLSocket cannot perform the handshake. I tried adding `startHandshake()` after the socket has been created in `SocketIO` and that fails. I'm not very familiare with the details of secure sockets, so I don't have any good idea for a fix.

I'm trying with these settings:
IMAP: mail.hvam.io:143
SMTP: smtp.hvam.io:587
and any random user and password, as we naver actually get to the authentification part.

This is the stacktrace:
```
javax.net.ssl.SSLHandshakeException: Handshake failed
        at com.android.org.conscrypt.ConscryptFileDescriptorSocket.startHandshake(ConscryptFileDescriptorSocket.java:286)
        at net.inbox.server.SocketIO.run(SocketIO.java:81)
        at java.lang.Thread.run(Thread.java:764)
     Caused by: javax.net.ssl.SSLProtocolException: SSL handshake aborted: ssl=0xe30b4888: Failure in SSL library, usually a protocol error
    error:100000f7:SSL routines:OPENSSL_internal:WRONG_VERSION_NUMBER (external/boringssl/src/ssl/tls_record.cc:242 0xe352ba43:0x00000000)
        at com.android.org.conscrypt.NativeCrypto.SSL_do_handshake(Native Method)
        at com.android.org.conscrypt.NativeSsl.doHandshake(NativeSsl.java:375)
        at com.android.org.conscrypt.ConscryptFileDescriptorSocket.startHandshake(ConscryptFileDescriptorSocket.java:224)
        	... 2 more
```

**Comment by itprojects** [comment]
SSL does not verify the host names. HTTPs does. Hence `'mail.hvam.io' != 'null'` the failure to give the correct name. One possibility is that host has a bad configuration, but it could also be a bug in Inbox Pager.

**Comment by itprojects** [comment]
`startHandshake()` when there is already an active SSL Session in progress, is supposed to result in a protocol error, by definition.

**Comment by itprojects** [comment]
@oizo Are there any changes if you use the following settings:

IMAP: mail.hvam.io:993
SMTP: smtp.hvam.io:465

**Comment by itprojects** [comment]
@oizo There is an experimental [bug-fix version.](https://github.com/itprojects/InboxPager/blob/apk-debug/InboxPager-461.apk)

Before you install it, you need to remove your old installation.
Then install the given apk.
Try imap, smtp. Always use ports 993 and 465, if you can.
Report here.

**Comment by itprojects** [comment]
@oizo If the link (https://github.com/itprojects/InboxPager/issues/30#issuecomment-625520592) in the above does not work, a new one can be made. Any luck?

**Comment by itprojects** [comment]
Fix in v5.0.
